src/main/java/org/spongepowered/api/CatalogType.java

```diff
@@ -25,8 +25,52 @@
 package org.spongepowered.api;
 
 /**
- * Represents a type of a catalog that can be used to identify types
- * without using an {@link Enum}.
+ * Represents a type of a catalog that can be used to identify types without
+ * using an {@link Enum}.
+ *
+ * <p>Implementing classes must meet the following requirements for all its
+ * members including those which are not listed in the classes listed in the
+ * {@link org.spongepowered.api.util.annotation.CatalogedBy CatalogedBy}
+ * annotation classes:
+ * <ul>
+ * <li>Every pair (a, b) that matches ...
+ * <ul>
+ * <li>a.equals(b) must also match a == b</li>
+ * <li>a.getId().equalsIgnoreCase(b.getId()) must also match a == b</li>
+ * <li>a.getName().equalsIgnoreCase(b.getName()) must also match a == b</li>
+ * </ul>
+ * </li>
+ * <li>The {@link GameRegistry#getType(Class, String)} method must serve all
+ * values independently from the case of the real id value.
+ * <ul>
+ * <li>It may but does not need to return the values for "well known aliases"
+ * </li>
+ * </ul>
+ * </li>
+ * <li>The value of a.getId() identifies the reference uniquely and must remain
+ * the same and valid.</li>
+ * <li>The value of a.getName() identifies the reference uniquely and must
+ * remain the same and valid.</li>
+ * <li>All methods that allow the combination of a CatalogType with other
+ * objects may not return a CatalogType.</li>
+ * </li>
+ * </ul>
+ * </p>
+ *
+ * <p>All Classes mentioned in the CatalogedBy annotation must meet the
+ * following requirements:</p>
+ * <ul>
+ * <li>The values referenced by catalog classes must remain the same and valid
+ * at all times.</li>
+ * <li>The variables in catalog classes may link to null if the given value is
+ * no longer valid and no appropriate alternative can be found. Thus two or more
+ * different variables can link to the same value. It is also possible that one
+ * or more values are not listed in the catalog classes.</li>
+ * <li>The variables in catalog classes may not be renamed or removed after
```

Comment:
In my opinion this isn't a very good requirement, a better one would be something requiring that removal of keys which are no longer valid must follow a sane deprecation policy.